pkg/secrets/censor.go

```diff
@@ -31,3 +34,23 @@ func (c *DynamicCensor) AddSecrets(s ...string) {
 	c.secrets.Insert(s...)
 	c.ReloadingCensorer.Refresh(c.secrets.List()...)
 }
+
+// ReadFromEnv loads content from the environment and adds it to the list.
```

Comment:
Why is this a method on `DynamicCensor` ?

Reply:
The idea was to centralize the methods for loading secrets to:

- reduce the code necessary in clients
- hopefully remind them that secrets should be censored

similar to `test-infra`'s secret `Agent` and `secretrecordingclient`.  Assuming that is useful, the choice between:

1. `secret, err := censor.ReadFromFile(path)`
2. `secret, err := secrets.ReadFromFile(path, &censor)`

was decided purely by character count.  I can't think of an advantage of the second form, but I'm also not opposed to it.

Reply:
Right, my comment was on the second form - I guess I prefer the composition there, it's weird for a secret recorder to close over file acces.